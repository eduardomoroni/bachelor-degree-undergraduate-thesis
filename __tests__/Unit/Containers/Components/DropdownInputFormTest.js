// @flow

import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
// $FlowFixMe
import AndroidDropDown from '../../../../src/Containers/Components/DropdownInputForm.android'
import iOSDropDown from '../../../../src/Containers/Components/DropdownInputForm.ios'
import { InputLabel, InputPicker } from '../../../../src/Containers/Components'

const props = {
  dropdownItems: ['item1', 'item2', 'item3'],
  selectedValue: 'item1',
  input: {
    name: 'beauty label',
    onChange: jest.fn() //  <-- This is a mock
  }
}

describe('<DropdownInputForm />', () => {

  const testDropdown = (component) => {
    const wrapper = shallow(component)
    const pickerProps = wrapper.find(InputPicker).props()
    console.log(wrapper.getNode())
    expect(wrapper.find(InputLabel).prop('label')).toEqual(props.input.name)
    expect(pickerProps.selectedValue).toEqual(props.selectedValue)
    expect(pickerProps.dropdownItems).toEqual(props.dropdownItems)
    expect(wrapper).toMatchSnapshot()
  }

  it('Should render android dropdown component', () => {
    testDropdown(<AndroidDropDown {...props} />)
    // const wrapper = shallow(<AndroidDropDown {...props} />)
    // const pickerProps = wrapper.find(InputPicker).props()

    // expect(wrapper.find(InputLabel).prop('label')).toEqual(props.input.name)
    // expect(pickerProps.selectedValue).toEqual(props.selectedValue)
    // expect(pickerProps.dropdownItems).toEqual(props.dropdownItems)
    // expect(wrapper).toMatchSnapshot()
  })

  it('Should render ios dropdown Component', () => {
    const props = { name: 'label', onChange: () => null }

    const wrapper = shallow(<iOSDropDown {...props} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.prop('onChange')).toEqual(props.onChange)
  })
})
