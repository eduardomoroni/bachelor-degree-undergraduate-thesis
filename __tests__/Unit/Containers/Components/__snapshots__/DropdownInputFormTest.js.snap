// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownInputForm /> Should render android dropdown component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgba(0, 0, 0, 0.4)",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-between",
      "marginBottom": 5,
    }
  }
>
  <InputLabel
    label="beauty label"
    onPress={[Function]}
  />
  <InputPicker
    dropdownItems={
      Array [
        "item1",
        "item2",
        "item3",
      ]
    }
    onValueChange={[Function]}
    selectedValue="item1"
  />
</View>
`;

exports[`<DropdownInputForm /> Should render ios dropdown Component 1`] = `
<iOSDropDown
  name="label"
  onChange={[Function]}
/>
`;
